lib/passport-saml/saml.js

```diff
@@ -837,7 +837,9 @@ SAML.prototype.hasValidSignatureForRedirect = function (container, originalQuery
 
 function validateSignatureForRedirect (urlString, signature, alg, cert) {
   var supportedAlgs = crypto.getHashes().filter(
-    function(h) { return new RegExp(h).test(alg); }
+    function(h) { return new RegExp(h, 'i').test(alg); }
```

Comment:
The "i" has been added here because URL encoding is not case sensitive, is that correct?

Reply:
no this is another issue. if azure is configured with `rsa-sha256` (lowercase) signing it would match with `sha256` crypto algorithm. doing a case incentive search in the crypto algorithms resolves this.
```js
crypto.getHashes()
[ 'RSA-MD4',
  'RSA-MD5',
  'RSA-MDC2',
  'RSA-RIPEMD160',
  'RSA-SHA1',
  'RSA-SHA1-2',
  'RSA-SHA224',
  'RSA-SHA256',
  'RSA-SHA384',
  'RSA-SHA512',
  'blake2b512',
  'blake2s256',
  'md4',
  'md4WithRSAEncryption',
  'md5',
  'md5-sha1',
  'md5WithRSAEncryption',
  'mdc2',
  'mdc2WithRSA',
  'ripemd',
  'ripemd160',
  'ripemd160WithRSA',
  'rmd160',
  'sha1',
  'sha1WithRSAEncryption',
  'sha224',
  'sha224WithRSAEncryption',
  'sha256',
  'sha256WithRSAEncryption',
  'sha384',
  'sha384WithRSAEncryption',
  'sha512',
  'sha512WithRSAEncryption',
  'ssl3-md5',
  'ssl3-sha1',
  'whirlpool' ]
```